### PR TITLE
Gherkin Scripts: Deploy Application using Templates

### PR DESCRIPTION
**Pull request by gajanan-more**
Fixes: https://issues.redhat.com/browse/ODC-5528

Problem: Gherkin Scripts for scenarios for application deployment using Templates provided by Red Hat

Solution: These are the gherkin scripts designed covering all the scenarios and types of applications user can deploy using Templates.

### CODE REVIEW

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        Background:
+            Given user is at developer perspective
+              And user is at Templates page
+
+
+        Scenario: Deploy CI/CD type Application using Templates
+            Given user has created or selected namespace "aut-addflow-templates"
+             When user selects CI/CD type
+              And user searches for "Jenkins"
+              And user clicks on "Jenkins" card
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:14]:
```suggestion
              And user searches and selects "Jenkins" card
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature [hunk 2]
```diff
@@ -0,0 +1,62 @@
+@add-flow
+Feature: Create Application using Templates
+              As a user, I want to deploy the application using Templates provided in Developer Catalog
+
+        Background:
+            Given user is at developer perspective
+              And user is at Templates page
+
+
+        Scenario: Deploy CI/CD type Application using Templates
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:10]:
```suggestion
        Scenario: Deploy CI/CD Application from Catalog Templates
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature (hunk 1 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:14]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature (hunk 2 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:10]: Updated as 
`Deploy CI/CD Application using Catalog Templates`

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature [hunk 3]
```diff
@@ -0,0 +1,62 @@
+@add-flow
+Feature: Create Application using Templates
+              As a user, I want to deploy the application using Templates provided in Developer Catalog
+
+        Background:
+            Given user is at developer perspective
+              And user is at Developer Catalog page
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:7]:
```suggestion
              And user is at Developer Catalog page
              And user has created or selected namespace "aut-addflow-templates"
              And user is at Templates page
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+
+        Scenario: Deploy CI/CD Application using Catalog Templates
+            Given user has created or selected namespace "aut-addflow-templates"
+             When user selects "Templates" option from Type section
+              And user selects CI/CD type
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:13]: Same given statement for all scenarios, can we move it to Background?
```suggestion
            When user selects CI/CD type
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+              And user clicks on Instantiate Template button
+              And user clicks on Create button
+             Then user will be redirected to Topology page
+              And user is able to see workload "jenkins" in topology page
+
+
+        Scenario: Deploy Database Application using Catalog Templates
+            Given user is at namespace "aut-addflow-templates"
+             When user selects "Templates" option from Type section
+              And user selects Databases type
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:24]: Same given statement for all scenarios, can we move it to Background?
```suggestion
            When user selects Databases type
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+              And user clicks on Instantiate Template button
+              And user clicks on Create button
+             Then user will be redirected to Topology page
+              And user is able to see workload "mariadb" in topology page
+
+
+        Scenario: Deploy Application using Languages Templates
+            Given user is at namespace "aut-addflow-templates"
+             When user selects "Templates" option from Type section
+              And user selects Languages type
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:35]:
```suggestion
              When user selects Languages type
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+              And user clicks on Instantiate Template button
+              And user clicks on Create button
+             Then user will be redirected to Topology page
+              And user is able to see workload "nodejs-postgresql-persistent" in topology page
+
+
+        Scenario: Deploy MiddleWare Application using Catalog Templates
+            Given user is at namespace "aut-addflow-templates"
+             When user selects "Templates" option from Type section
+              And user selects MiddleWare type
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:46]:
```suggestion
              When user selects MiddleWare type
```

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+              And user clicks on Instantiate Template button
+              And user clicks on Create button
+             Then user will be redirected to Topology page
+              And user is able to see workload "httpd-example" in topology page
+
+
+        Scenario: Deploy Application using Other types of Catalog Templates
+            Given user is at namespace "aut-addflow-templates"
+             When user selects "Templates" option from Type section
+              And user selects Other type
```
Comment (makambalaji) [inline frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:57]:
```suggestion
              When user selects Other type
```

**Review by makambalaji — CHANGES_REQUESTED** [review]
Removed the duplicate lines and added them in Background section

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature (hunk 3 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:7]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature (hunk 4 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:13]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature (hunk 5 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:24]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature (hunk 6 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:35]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature (hunk 7 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:46]: Updated

File: frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature (hunk 8 above)
Comment (gajanan-more) [reply frontend/packages/dev-console/integration-tests/features/addFlow/create-from-templates.feature:57]: Updated

**Pull request merged**